File: module_b/src/main/java/com/github/xiaogegechen/module_b/view/FragmentB.java
package com.github.xiaogegechen.module_b.view;

import android.view.View;
import android.widget.GridView;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.github.xiaogegechen.common.base.BaseFragment;
import com.github.xiaogegechen.common.util.LogUtil;
import com.github.xiaogegechen.module_b.R;
import com.github.xiaogegechen.module_b.adapter.ConstellationAdapter;
import com.github.xiaogegechen.module_b.model.Constellation;
import com.github.xiaogegechen.module_b.model.Params;
import com.github.xiaogegechen.module_b.presenter.FragmentBPresenterImpl;
import com.github.xiaogegechen.module_b.presenter.IFragmentBPresenter;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import static com.github.xiaogegechen.common.arouter.ARouterMap.MODULE_B_FRAGMENT_B_PATH;

@Route(path = MODULE_B_FRAGMENT_B_PATH)
public class FragmentB extends BaseFragment implements IFragmentBView{

    private static final String TAG = "FragmentB";

    private GridView mGridView;
    private IFragmentBPresenter mFragmentBPresenter;

    @Override
    public void initView(@NotNull View view) {
        mGridView = view.findViewById(R.id.module_b_grid_view);
    }

    @Override
    public void initData() {
        mFragmentBPresenter = new FragmentBPresenterImpl(obtainContext());
        mFragmentBPresenter.attach(this);
        List<Constellation> constellationList = new ArrayList<>();
        // TODO 以资源引用的方式使用这些字符串
        constellationList.add(new Constellation(R.drawable.baiyangzuo, "白羊座","3.21~4.19"));
        constellationList.add(new Constellation(R.drawable.jinniuzuo, "金牛座","4.20~5.20"));
        constellationList.add(new Constellation(R.drawable.shuangzizuo, "双子座","5.21~6.21"));
        constellationList.add(new Constellation(R.drawable.juxiezuo, "巨蟹座","6.22~7.22"));
        constellationList.add(new Constellation(R.drawable.shizizuo, "狮子座","7.23~8.22"));
        constellationList.add(new Constellation(R.drawable.chunvzuo, "处女座","8.23~9.22"));
        constellationList.add(new Constellation(R.drawable.tianchengzuo, "天秤座","9.23~10.23"));
        constellationList.add(new Constellation(R.drawable.tianxiezuo, "天蝎座","10.24~11.22"));
        constellationList.add(new Constellation(R.drawable.sheshouzuo, "射手座","11.23~12.21"));
        constellationList.add(new Constellation(R.drawable.mojiezuo, "摩羯座","12.22~1.19"));
        constellationList.add(new Constellation(R.drawable.shuipingzuo, "水瓶座","1.20~2.18"));
        constellationList.add(new Constellation(R.drawable.shuangyuzuo, "双鱼座","2.19~3.20"));
        mGridView.setAdapter(new ConstellationAdapter(constellationList, obtainContext()));
        mGridView.setOnItemClickListener((parent, view, position, id) -> {
            LogUtil.d(TAG, "mGridView is click, position is: " + position);
            Constellation constellation = (Constellation) parent.getAdapter().getItem(position);
            // 传递星座名和iconId
            String consName = constellation.getName();
            int iconId = constellation.getIconId();
            Params params = new Params(consName, iconId);
            mFragmentBPresenter.gotoDetailActivity(params);
        });
    }

    @Override
    public void onDestroy() {
        mFragmentBPresenter.detach();
        super.onDestroy();
    }

    @Override
    public int getLayoutId() {
        return R.layout.module_b_fragment_b;
    }

    @Override
    public void showProgress() {

    }

    @Override
    public void showErrorPage() {

    }

    @Override
    public void showToast(@NotNull String message) {

    }
}

File: module_b/src/main/java/com/github/xiaogegechen/module_b/view/FragmentB.kt
package com.github.xiaogegechen.module_b.view

import android.annotation.SuppressLint
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.Toast
import com.alibaba.android.arouter.facade.annotation.Route
import com.github.xiaogegechen.common.arouter.ARouterMap.Companion.MODULE_B_FRAGMENT_B_PATH
import com.github.xiaogegechen.common.base.BaseFragment
import com.github.xiaogegechen.common.util.ImageParam
import com.github.xiaogegechen.common.util.ImageUtil
import com.github.xiaogegechen.module_b.R
import com.github.xiaogegechen.module_b.model.Today
import com.github.xiaogegechen.module_b.model.Week
import com.github.xiaogegechen.module_b.model.Year
import com.github.xiaogegechen.module_b.presenter.FragmentBPresenterImpl
import kotlinx.android.synthetic.main.module_b_fragment_b.*
import org.angmarch.views.NiceSpinner

@Route(path = MODULE_B_FRAGMENT_B_PATH)
class FragmentB: BaseFragment(),IFragmentBView {

    private var mFragmentBPresenter: FragmentBPresenterImpl? = null
    private var mBgImage: ImageView? = null
    private lateinit var name: String

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)

        val spinner: NiceSpinner? = view?.findViewById(R.id.spinner)
        mBgImage = view?.findViewById(R.id.imageBg)
        spinner?.attachDataSource(CONSTELLATION.asList())
        spinner?.addOnItemClickListener { _, _, _, _ ->
            name = spinner.selectedItem as String
            mFragmentBPresenter?.queryToday(name)
            mFragmentBPresenter?.queryWeek(name)
            mFragmentBPresenter?.queryYear(name)
        }

        mFragmentBPresenter?.attach(this)
        mFragmentBPresenter?.queryImage()
        mFragmentBPresenter?.queryToday(name)
        mFragmentBPresenter?.queryWeek(name)
        mFragmentBPresenter?.queryYear(name)

        return view
    }

    override fun initData() {
        mFragmentBPresenter = FragmentBPresenterImpl()
        name = "摩羯座"
    }

    override fun getLayoutId(): Int {
        return R.layout.module_b_fragment_b
    }

    override fun onDestroy() {
        super.onDestroy()
        mFragmentBPresenter?.detach()
    }

    override fun showToady(today: Today) {
        todayTitle.name?.text = today.date
        todayTitle.value?.text = today.name
        todayAll.value?.text = today.all
        todayColor.value?.text = today.color
        todayHealth.value?.text = today.health
        todayLove.value?.text = today.love
        todayMoney.value?.text = today.money
        todayLuckyNum.value?.text = today.number
        todayFriend.value?.text = today.friend
        todayWork.value?.text = today.work
        todaySum.text = today.summary
    }

    @SuppressLint("SetTextI18n")
    override fun showWeek(week: Week) {
        weekTitle.value?.text = "${week.name}本周运势"
        weekTitle.name?.text = week.date
        weekNum.text = "本周是今年的第${week.weekth}周"
        weekLove.text = "${week.love}"
        weekMoney.text = "${week.money}"
        weekWork.text = "${week.work}"
    }

    @SuppressLint("SetTextI18n")
    override fun showYear(year: Year) {
        yearTitle.value?.text = "${year.name}今年运势"
        yearTitle.name?.text = "今年是${year.year}年"
        yearPwd.text = "年度密码：${year.mima?.info}"
        yearSum.text = "年度概述：${year.mima?.text?.get(0)}"
        yearWork.text = "事业运：${year.career?.get(0)}"
        yearLove.text = "感情运：${year.love?.get(0)}"
        yearHealth.text = "健康运：${year.health?.get(0)}"
        yearMoney.text = "财运：${year.finance?.get(0)}"
    }

    override fun showBgImage(url: String) {
        ImageUtil.displayImage(ImageParam.Builder()
            .context(obtainContext())
            .error(ColorDrawable(obtainResources()?.getColor(R.color.design_color_accent)!!))
            .placeholder(ColorDrawable(obtainResources()?.getColor(R.color.design_color_accent)!!))
            .imageView(mBgImage)
            .url(url)
            .build())
    }

    override fun showProgress() {
    }

    override fun showErrorPage() {
    }

    override fun showToast(message: String) {
        Toast.makeText(obtainContext(), message, Toast.LENGTH_SHORT).show()
    }

    companion object{
        private val CONSTELLATION = arrayOf("摩羯座","白羊座","金牛座","双子座","巨蟹座","狮子座","处女座","天秤座","天蝎座","射手座","水瓶座","双鱼座")
    }
}